src/main/java/org/skriptlang/skript/bukkit/loottables/LootTableUtils.java

```diff
@@ -0,0 +1,57 @@
+package org.skriptlang.skript.bukkit.loottables;
+
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockState;
+import org.bukkit.loot.LootTable;
+import org.bukkit.loot.Lootable;
+
+/**
+ * Utility class for loot tables.
+ */
+public class LootTableUtils {
+
+	/**
+	 * * Checks whether a block or entity is an instance of {@link Lootable}. This is done because a block is not an instance of Lootable, but a block state is.
+	 * @param object the object to check.
+	 * @return whether the object is lootable.
+	 */
+	public static boolean isLootable(Object object) {
+		if (object instanceof Block block)
+			object = block.getState();
+		return object instanceof Lootable;
+	}
+
+	/**
+	 * Gets the Lootable instance of an object. You should call {@link #isLootable(Object)} before calling this method.
+	 * @param object the object to get the Lootable instance of.
+	 * @return the Lootable instance of the object.
+	 */
+	public static Lootable getAsLootable(Object object) {
+		if (object instanceof Block block)
+			object = block.getState();
+		if (object instanceof Lootable lootable)
+			return lootable;
+		return null;
+	}
+
+	/**
+	 * Gets the loot table of an object. You should call {@link #isLootable(Object)} before calling this method.
+	 * @param object the object to get the loot table of.
+	 * @return returns the LootTable of the object.
+	 */
+	public static LootTable getLootTable(Object object) {
```

Comment:
```suggestion
	public static @Nullable LootTable getLootTable(Object object) {
```

Reply:
we talked about how we wouldn't do this in a previous conversation

Reply:
oh ok, then just ignore it.